irida_staramr_results/cli.py

```diff
@@ -14,22 +18,90 @@ def init_argparser():
     )
     argument_parser.add_argument("-v", "--version", action="version", version=f"{argument_parser.prog} {__version__}",
                                  help="The current version of irida-staramr-results.")
-    argument_parser.add_argument("-p", "--project", action="store",
+    argument_parser.add_argument("-p", "--project", action="store", required=True, type=int,
                                  help="Project(s) to scan for StarAMR results.")
-    argument_parser.add_argument("-o", "--output", action="store",
+    argument_parser.add_argument("-o", "--output", action="store", default="out.xlsx",
                                  help="The name of the output excel file.")
     argument_parser.add_argument("-u", "--username", action="store",
                                  help="This is your IRIDA account username.")
     argument_parser.add_argument("-pw", "--password", action="store",
                                  help="This is your IRIDA account password.")
+    argument_parser.add_argument("-c", "--config", action='store', default=DEFAULT_CONFIG_PATH,
+                                 help='Path to an alternative configuration file. '
+                                      'This overrides the default config file in the config directory')
 
     return argument_parser
 
 
+def _validate_args(args):
+    """
+    Validates argument input by the users and returns a dictionary of required information from arguments.
+    If user does not include username and password in arguments,
+    the program prompts the user to enter it.
+    :param args:
+    :return dictionary:
+    """
+    if args.username is None:
+        args.username = input("Enter your IRIDA username: \n")
+    if args.password is None:
+        print("Enter your IRIDA password: ")
+        args.password = getpass.getpass()
+
+    return {'username': args.username,
+            'password': args.password,
+            'config': args.config,
+            'project': args.project,
+            'output': args.output}
+
+
+def _parse_config(config_file_path):
```

Comment:
I think we should expect the full path to a file, as not to assume that the file will be named `config.yml`

Reply:
Done.